lesson09/script.js

```diff
@@ -0,0 +1,46 @@
+function showLoadingImg(){
+    const loadingImg = document.createElement('img');
+    loadingImg.src = "loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+function showCategories(result){
+    const fragment = document.createDocumentFragment();
+    for(const list of result){
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+        a.textContent = list.text;
+        a.href = `/${list.to}`;
+        img.src = list.img;
+        img.alt = list.alt;
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    document.getElementById('js-list').appendChild(fragment);
+}
+
+function SetCategoriesAfter3Seconds(){
+    return new Promise(resolve => {
+    const categories = [
+        {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+        {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+    ];
+    setTimeout(() => {
+        resolve(categories);
+        },3000);
+    });
+}
+
+async function awaitGetCategories(){
+    let result = await SetCategoriesAfter3Seconds();
```

Comment:
Q.
Why did you use let here? Is there any reasons?

Reply:
This is a mistake.
Thank you for teaching me.
This variable is a variable that does not need to be reassigned, so const is appropriate.
I will fix it.